### PR TITLE
Cleanup of dispatcher code

### PR DESCRIPTION
**Pull request by stojg**
This will extend the environment config editor to use the base dispatcher and contains a couple
of minor cleanups
- check read access to project and env in init
- keep a in memory copy of project and env after they have been fetched in init

### CODE REVIEW

File: code/EnvironmentConfig/Dispatcher.php [hunk 1]
```diff
@@ -43,35 +59,40 @@ public function init() {
...
 	}
 
 	/**
+	 * @param string $name
+	 *
+	 * @return array
+	 */
+	public function getModel($name) {
+		$model = [
+			'InitialSecurityID' => $this->getSecurityToken()->getValue(),
```
Comment (mateusz) [inline code/EnvironmentConfig/Dispatcher.php]: This shouldn't be needed. \Dispatcher adds it automatically.

**Comment by mateusz** [comment]
You should be able to remove the tools.jsx:readInlineData function and rely on deploynaut/js/tools.jsx (which should be pullable via browserify across modules)

**Comment by mateusz** [comment]
You should be able to replace the code in platform.jsx with something like

```
var Tools = require('../../deploynaut/js/tools.jsx');

var Variables = require('./Variables.jsx');
Tools.install(Variables, 'Variables');
```

**Comment by mateusz** [comment]
You also should redo the `EnvironmentConfig_configuration.ss` to use `$getReactComponent(Variables)` instead of the copy-pasted boilerplate. This will force you to get environmentConfigContext from a global in Variables.jsx (as opposed to explicitly passed to a component). Which is bad. Which is what I've done in deploynaut-usermanagement.

**Comment by stojg** [comment]
I agree on the jsx stuff, currently rewriting it

**Comment by mateusz** [comment]
Also, I have - kind of - standardised on calling the jsx "React component" files with UpperCamelCase. Because then it nicely ties the three elements: the ss boilerplate, the jsx boilerplate and the php boilerplate.

**Comment by mateusz** [comment]
Have a read through https://github.com/silverstripe/deploynaut/blob/master/docs/dispatcher.md

File: code/EnvironmentConfig/Dispatcher.php (hunk 1 above)
Comment (stojg) [inline code/EnvironmentConfig/Dispatcher.php]: true, I forgot that.

**Comment by mateusz** [comment]
Maybe renaming the Dispatcher to VariableDispatcher or ConfigurationDispatcher would be A Good Thing® too, so the naming matches and we have 1 dispatcher per 1 react component.

**Pull request merged**